bolero/representation/dmp_sequence.py

```diff
@@ -323,3 +323,65 @@ def can_step(self):
         """Returns true if step() can be called again, false otherwise."""
         return len(np.where(self.steps <= self.split_steps)[0]) > 0
 
+    def trajectory(self):
+        """Generate trajectory represented by the DMP in open loop.
+
+        The function can be used for debugging purposes.
+
+        Returns
+        -------
+        X : array, shape (n_steps, n_task_dims)
+            Positions
+
+        Xd : array, shape (n_steps, n_task_dims)
+            Velocities
+
+        Xdd : array, shape (n_steps, n_task_dims)
+            Accelerations
+        """
+        last_t = 0.0
+        last_y = np.copy(self.subgoals[0])
+        last_yd = np.copy(self.subgoal_velocities[0])
+        last_ydd = np.zeros(self.n_task_dims)
+
+        y = np.empty(self.n_task_dims)
+        yd = np.empty(self.n_task_dims)
+        ydd = np.empty(self.n_task_dims)
+
+        Y = []
```

Comment:
I think we can pre-allocate the complete numpy arrays here already.

Reply:
Pre-allocated the trajectory output arrays :+1: